src/averageAges.js

```diff
@@ -15,11 +15,17 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => {
+    return century !== undefined
+      ? person.sex === 'm' && century === Math.ceil(person.died / 100)
+      : person.sex === 'm';
+  });
+
+  const totalAge = men.reduce((sum, item) => {
+    return sum + (item.died - item.born);
+  }, 0);
+
+  return +(totalAge / men.length).toFixed(2);
```

Comment:
You can replace `+(totalAge / men.length).toFixed(2);` with `Number((totalAge / men.length).toFixed(2));` for better readability.